src/moonlink/src/storage/async_bitwriter.rs

```diff
@@ -0,0 +1,295 @@
+use more_asserts as ma;
+use std::io;
+use tokio::io::{AsyncWrite, AsyncWriteExt};
+use tokio_bitstream_io::{BitQueue, Endianness, Numeric};
+
+/// Buffer size, which controls the flush threshold.
+const BUFFER_SIZE: usize = 4096;
+
+/// [`BitWriter`] implement asynchronous write for bits; to avoid excessive runtime rescheduling caused by `await`,
```

Comment:
I found my latest doc change is not pushed successfully, will followup:
```rs
/// [`BitWriter`] implement asynchronous write for bits;
/// - To avoid excessive runtime rescheduling caused by `await`, it takes cooperative interface, which means it requires users to explicit flush.
/// - To avoid excessive syscalls, bits are buffered in-memory until it reaches flush threshold.
```